package/com.unity.formats.usd/Runtime/Scripts/IO/Geometry/MeshExporter.cs

```diff
@@ -176,8 +176,8 @@ public static class MeshExporter {
         Debug.LogWarning("Null mesh for: " + path);
         return;
       }
-      if (mesh.isReadable == false) {
-        Debug.LogWarning("Mesh not readable: " + objContext.path);
+      if (mesh.isReadable == false && !Application.isEditor) {
```

Comment:
Are you sure this is true? I believe even in-editor meshes can be non-readable at run-time.

Reply:
Seems we are both right and there's actually 3 cases for the isReadable bool. Fancy quantum logic I guess.
(See https://docs.unity3d.com/ScriptReference/Mesh-isReadable.html).

1. it always works in Edit Mode
2. it works in Play Mode when **not** accessed from inside the game loop (accessed from Editor works)
3. it does not work in Play Mode when accessed from inside the game loop (this matches behaviour in build).

I updated the code to properly check for that. Unfortunately the "canAccess" property on a mesh, which controls that behaviour, is internal`, so I had to use reflection - not ideal.
A better check would be some kind of "Application.isInsideGameLoop". Are you aware of something like that?

Reply:
Wow, interesting. I don't know off hand, but I'll ask around.